code/modules/tables/interactions.dm

```diff
@@ -57,6 +57,65 @@
 			return 1
 	return 1
 
+/obj/structure/table/Crossed(var/atom/movable/am as mob|obj)
+	..()
+	if(ishuman(am))
+		var/mob/living/carbon/human/H = am
+		if(H.a_intent != I_HELP || H.m_intent == "run")
+			throw_things(H)
+		else if(H.is_diona() || H.species.bodytype == "Heavy machine")
+			throw_things(H)
+	else if((isliving(am) && !issmall(am)) || isslime(am))
+		var/mob/living/L = am
+		throw_things(L)
+
+/obj/structure/table/proc/throw_things(var/mob/living/user = null)
+	var/list/targets = list(get_step(src,dir),get_step(src,turn(dir, 45)),get_step(src,turn(dir, -45)))
+	var/turf/T = get_turf(src)
+	var/list/things = T.contents
+	var/obj/item/I = (/obj/item in things)
+	if(!I)
+		return
+	for (I in things)
+		spawn(0)
+			I.throw_at(pick(targets),1,1)
+	if(user)
+		usr.visible_message("<span class='warning'>[user] knocks everything from \the [src]!</span>")
+
+
+/obj/structure/table/structure_shaken()
+	..()
+	throw_things()
+
+
+/obj/structure/table/do_climb(var/mob/living/user)

```

Comment:
Did you just copy the base definition up here?

Reply:
Yes I did